pkg/client/cache/listwatch.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/watch"
+)
+
+// ListWatch knows how to list and watch a set of apiserver resources.  It satisfies the ListerWatcher interface.
+// It is a convenience function for users of NewReflector, etc.
+type ListWatch struct {
+	Client        *client.Client
+	FieldSelector labels.Selector
+	Resource      string
+}
+
+// ListWatch knows how to list and watch a set of apiserver resources.
+func (lw *ListWatch) List() (runtime.Object, error) {
+	return lw.Client.
+		Get().
```

Comment:
Sure.

```shell
kubelet --manifest_url=http://metadata.google.internal/computeMetadata/v1beta1/instance/attributes/google-container-manifest --config=/etc/kubernetes/manifests
```

You can reproduce this easily by starting a new containervm instance:

```shell
gcloud compute instances create --image container-vm ... --metadata-from-file google-container-manifest=<yamlfile>.yaml
```

This is roughly the YAML snippet I'm using, I'm fairly certain you'll reproduce it with this:
https://cloud.google.com/compute/docs/containers/container_vms#create_the_manifest

Then SSH to your image, replace the kubelet binary with the one from head, edit /etc/default/kubelet to replace the flags with a single dash with two dashes (required since the switch to pflag which uses POSIX-like flags) and reboot it, you'll see this error.

But then, you might be able to reproduce it outside of GCE as well...

Thanks for taking a look at this! Ping me if you have questions for me.

Cheers,
Filipe